src/vs/base/node/stream.ts

```diff
@@ -96,4 +96,70 @@ export function readExactlyByFile(file: string, totalBytes: number, callback: (e
 
 		loop();
 	});
+}
+
+/**
+ * Reads a file until a matching string is found.
+ *
+ * @param file The file to read.
+ * @param matchingString The string to search for.
+ * @param chunkBytes The number of bytes to read each iteration.
+ * @param maximumBytesToRead The maximum number of bytes to read before giving up.
+ * @param callback The finished callback.
+ */
+export function readToMatchingString(file: string, matchingString: string, chunkBytes: number, maximumBytesToRead: number, callback: (error: Error, result: string) => void): void {
+	fs.open(file, 'r', null, (err, fd) => {
+		if (err) {
+			return callback(err, null);
+		}
+
+		function end(err: Error, result: string): void {
+			fs.close(fd, (closeError: Error) => {
+				if (closeError) {
+					return callback(closeError, null);
+				}
+
+				if (err && (<any>err).code === 'EISDIR') {
+					return callback(err, null); // we want to bubble this error up (file is actually a folder)
+				}
+
+				return callback(null, result);
+			});
+		}
+
+		let buffer = new Buffer(maximumBytesToRead);
+		let bytesRead = 0;
+		let zeroAttempts = 0;
+		function loop(): void {
+			fs.read(fd, buffer, bytesRead, chunkBytes, null, (err, moreBytesRead) => {
+				if (err) {
+					return end(err, null);
+				}
+
+				// Retry up to N times in case 0 bytes where read
+				if (moreBytesRead === 0) {
+					if (++zeroAttempts === 10) {
+						return end(null, null);
+					}
+
+					return loop();
+				}
+
+				bytesRead += moreBytesRead;
+
+				const newLineIndex = buffer.indexOf(matchingString);
+				if (buffer.indexOf('\n') >= 0) {
```

Comment:
@Tyriar is this an oversight? why suddenly you look for `\n` and call it `newLineIndex` if `matchingString` can be anything?